src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessModuleTests.cs

```diff
@@ -18,26 +19,30 @@ public void TestModulePropertiesExceptOnOSX()
             if (global::Interop.IsWindows)
                 fileName = string.Format("{0}.exe", CoreRunName);
 
-            // Ensure the process has loaded the modules.
-            Assert.True(SpinWait.SpinUntil(() =>
+            try
             {
-                if (_process.Modules.Count > 0)
-                    return true;
-                _process.Refresh();
-                return false;
-            }, WaitInMS));
+                // Ensure the process has loaded the modules.
+                Assert.True(SpinWait.SpinUntil(() =>
+                {
+                    if (_process.Modules.Count > 0)
+                        return true;
+                    _process.Refresh();
+                    return false;
+                }, WaitInMS));
 
-            ProcessModuleCollection moduleCollection = _process.Modules;
-            Assert.True(moduleCollection.Count > 0);
+                ProcessModuleCollection moduleCollection = _process.Modules;
+                Assert.True(moduleCollection.Count > 0);
 
-            ProcessModule coreRunModule = _process.MainModule;
-            Assert.True(coreRunModule.BaseAddress.ToInt64() > 0);
-            Assert.True(coreRunModule.EntryPointAddress.ToInt64() >= 0);
-            Assert.Equal(fileName, coreRunModule.ModuleName);
-            Assert.True(coreRunModule.ModuleMemorySize > 0);
-            Assert.EndsWith(fileName, coreRunModule.FileName);
+                ProcessModule coreRunModule = _process.MainModule;
+                Assert.True(coreRunModule.BaseAddress.ToInt64() > 0);
+                Assert.True(coreRunModule.EntryPointAddress.ToInt64() >= 0);
+                Assert.Equal(fileName, coreRunModule.ModuleName);
+                Assert.True(coreRunModule.ModuleMemorySize > 0);
+                Assert.EndsWith(fileName, coreRunModule.FileName);
 
-            Assert.Equal(string.Format("System.Diagnostics.ProcessModule ({0})", fileName), coreRunModule.ToString());
+                Assert.Equal(string.Format("System.Diagnostics.ProcessModule ({0})", fileName), coreRunModule.ToString());
+            }
+            catch (Win32Exception) { } // expected, known behavior of Process.MainModule to throw, when 32 bit process accesses a 64bit PEB on 64 bit machines.
```

Comment:
No. The underlying implementation calls EnumProcessModules (https://msdn.microsoft.com/en-us/library/windows/desktop/ms682631(v=vs.85).aspx) which has the following limitation.

*If this function is called from a 32-bit application running on WOW64, it can only enumerate the modules of a 32-bit process. If the process is a 64-bit process, this function fails and the last error code is ERROR_PARTIAL_COPY (299).*

The problem can be partially solved by using EnumProcessModulesEx (https://msdn.microsoft.com/en-us/library/windows/desktop/ms682633(v=vs.85).aspx) pending AppCompat bar-check however I would like revisit these APIs holistically due to the flakiness and cross-plat issues in general.